### PR TITLE
[posix] add radio URL parameter cca-threshold

### PR DESCRIPTION
**Pull request by Moandor-y**
Fixes <https://github.com/openthread/ot-br-posix/issues/609>.

### CODE REVIEW

File: src/posix/platform/radio.cpp [hunk 1]
```diff
@@ -96,10 +96,11 @@ void otPlatRadioSetPromiscuous(otInstance *aInstance, bool aEnable)
...
 {
-    ot::Posix::RadioUrl &radioUrl       = *static_cast<ot::Posix::RadioUrl *>(aRadioUrl);
-    bool                 resetRadio     = (radioUrl.GetValue("no-reset") == nullptr);
-    bool                 restoreDataset = (radioUrl.GetValue("ncp-dataset") != nullptr);
-    const char *         urlFemLnaGain  = radioUrl.GetValue("fem-lnagain");
+    ot::Posix::RadioUrl &radioUrl        = *static_cast<ot::Posix::RadioUrl *>(aRadioUrl);
+    bool                 resetRadio      = (radioUrl.GetValue("no-reset") == nullptr);
+    bool                 restoreDataset  = (radioUrl.GetValue("ncp-dataset") != nullptr);
+    const char *         urlFemLnaGain   = radioUrl.GetValue("fem-lnagain");
+    const char *         urlCcaThreshold = radioUrl.GetValue("ccathreshold");
```
Comment (simonlingoogle) [inline src/posix/platform/radio.cpp:103]:
```suggestion
    const char *         urlCcaThreshold = radioUrl.GetValue("cca-threshold");
```
nit: follow the same naming convention of other parameters to use `-` to separate words?

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍 

with a small suggestoin

File: src/posix/platform/radio.cpp (hunk 1 above)
Comment (Moandor-y) [reply src/posix/platform/radio.cpp]: But in CLI it's one word `ccathreshold`. Should we keep consistent about this?

File: src/posix/platform/radio.cpp [hunk 2]
```diff
@@ -99,7 +99,7 @@ void platformRadioInit(otUrl *aRadioUrl)
     ot::Posix::RadioUrl &radioUrl       = *static_cast<ot::Posix::RadioUrl *>(aRadioUrl);
     bool                 resetRadio     = (radioUrl.GetValue("no-reset") == nullptr);
     bool                 restoreDataset = (radioUrl.GetValue("ncp-dataset") != nullptr);
-    const char *         urlFemLnaGain  = radioUrl.GetValue("fem-lnagain");
+    const char *         urlValue;
```
Comment (bukepo) [inline src/posix/platform/radio.cpp:102]: Prefer `parameterValue` or `value` to `urlValue`.
```suggestion
    const char *         value;
```

**Review by bukepo — APPROVED** [review]
LGTM:+1:

File: src/posix/platform/radio.cpp (hunk 2 above)
Comment (Moandor-y) [reply src/posix/platform/radio.cpp]: Changed to `parameterValue`.

File: src/posix/platform/radio.cpp (hunk 1 above)
Comment (simonlingoogle) [reply src/posix/platform/radio.cpp]: Just a weak recommendation that radio URL parameters keep consistent with each other.

Comment (jwhui) [reply src/posix/platform/radio.cpp]: I have a slight preference to maintain consistency among URL parameters.

Comment (Moandor-y) [reply src/posix/platform/radio.cpp]: Updated to `cca-threshold`.

**Review by jwhui — APPROVED** [review]

**Pull request merged**